### ISSUE TITLE
Project 2 : Explore the recognition performance of YOLO model for fogging images 

### ISSUE THREAD

**Issue by htyaamazing** [htyaamazing/Blog#3]
### Tianyu He  G49072865
In recent years, the progress of science and technology has brought more and more convenience to human daily life, and with the development of artificial intelligence technology, automatic driving technology has also become a reality. With the number of vehicles increasing year by year, the frequency of traffic accidents is also increasing. Smog and other weather reasons have become one of the main causes of traffic accidents. In order to reduce the occurrence of traffic accidents and protect the life safety of passengers, the research of automatic driving system has become a general trend.

This blog uses darknet, a version of Windows written by AlexeyAB, to build a network model test. At the same time, the open driving data set BDD100K released by the AI Laboratory of Berkeley University (BAIR) is used to preprocess the images of the data set by using the fog algorithm, and the image results detected in foggy weather are simulated. The ability of the model to process noisy images is also explored.

Because the BDD100K data set is too large, there are nearly 100,000 high-definition videos, the 10th second of each video to sample the key frame, to get 100,000 pictures. So I chose just a few of these images to test.

Here are samples of pictures: 
<p align="center">
<img src="https://github.com/htyaamazing/Blog/assets/82633085/946399b8-aa03-4660-b444-204d74e6563e" >
</p>

### Image processing
First, I picked two sets of images to process with analog fog and analog noise.As for why I did this, the main reason is that citations so that the model's performance in complex environments can be explored: real-world driving environments are very complex and variable, including different weather conditions (e.g., fog, rain, snow) and lighting conditions (e.g., bright light, dim light). By adding simulated fog and noise to the training data, you can help the model learn to recognize objects correctly in these complex environments.

Here, I used ChatGPT to help me write two scripts for noise processing of images with adjustable parameters.Two sets of pictures were generated by passing each set through noise processing levels 1 to 10. 

**The thumbnails of the images are shown below.**

<p align="center">
<img src="https://github.com/htyaamazing/Blog/assets/82633085/b47064e8-0bea-43fe-9ee5-9616f16963b4" height = "400" width = "600">
</p>

This is the second set of pictures:

<p align="center">
<img src="https://github.com/htyaamazing/Blog/assets/82633085/169dda7e-6849-4e21-9632-fff7599ff05f" height = "400" width = "600">
</p>

### Experimental section

Now let's test the recognition effect of the processed picture.

First, I chose darknet's version of yolov3, which was easier to use because of my previous experience with him.I chose darknet's version of yolov3, which was easier to use because of my previous experience with him. Next, I used the processed images one by one to recognize and classify them using the classifier, here are some examples.

<p align="center">
<img src="https://github.com/htyaamazing/Blog/assets/82633085/a88538da-c539-4ea2-a412-514f67fa3799" height = "400" width = "600">
</p>

We can see. The classifier is still more accurate in recognizing images when the noise is not yet very large.But what about when we turn up the noise parameter a bit?

<p align="center">
<img src="https://github.com/htyaamazing/Blog/assets/82633085/35c6d220-a324-4d7b-9b01-56a2bb8096f2" height = "400" width = "600">
</p>

We can clearly see that while the near objects can still be recognized, the far ones are already almost unrecognizable.
So, I recorded and listed an Excel sheet and line graph based on the results of each this treatment.

<p align="center">
<img src="https://github.com/htyaamazing/Blog/assets/82633085/73793231-7fe3-4bfa-b0a8-f0b56823b3cc" height = "300" width = "400">
</p>

As you can see inside this chart, both CAR and PERSON recognition decreases gradually as the fog level increases. This is very much in line with our normal perception. However, the other noise-processed image curve presents some problems.

<p align="center">
<img src="https://github.com/htyaamazing/Blog/assets/82633085/da8c0802-4c36-41a6-b3f2-db8530f2f2bc" height = "300" width = "400">
</p>

We can see that while TRUCK does decrease overall as the noise level decreases, the recognition accuracy of CAR does not indeed decrease, but keeps floating around 94%.

<p align="center">
<img src="https://github.com/htyaamazing/Blog/assets/82633085/8ca72ad3-1649-4186-8e5c-c11482d893db" height = "300" width = "500">
</p>

As you can see from the above example, even though the noise level is already very high and the distant truck can no longer be recognized, the recognition of the near car is still very good.

**So WHY IS THAT?**

After doing some research, I think there are several reasons for this.

**Robustness of feature extraction**: YOLOv3 and other deep learning models are designed with robustness of feature extraction in mind. The model extracts features from the image through a multi-layer convolutional neural network, ranging from simple edges and textures to complex object parts. Nearby car occupies more pixels and provides richer feature information, and can still be accurately recognized by these features even in the presence of more noise.

**Contextual information**: the YOLOv3 model utilizes the contextual information of the entire image for object recognition. This means that even if some parts of the image are blurred due to noise, the model is able to infer the class of the object based on the surrounding environment and contextual clues. Nearby car is more likely to provide enough contextual information for the model to recognize due to its larger size.

**Object size and resolution**: Nearby objects occupy more pixels in the image, so the key features of these objects (e.g., edges, shape, color, etc.) remain relatively legible even with added noise. In contrast, distant trucks have lower resolution in the image due to their smaller size, and noise is more likely to mask their key features, leading to recognition difficulties.

After doing some research, I think there are several reasons for this.

Robustness of feature extraction: YOLOv3 and other deep learning models are designed with robustness of feature extraction in mind. The model extracts features from the image through a multi-layer convolutional neural network, ranging from simple edges and textures to complex object parts. Nearby car occupies more pixels and provides richer feature information, and can still be accurately recognized by these features even in the presence of more noise.

Contextual information: the YOLOv3 model utilizes the contextual information of the entire image for object recognition. This means that even if some parts of the image are blurred due to noise, the model is able to infer the class of the object based on the surrounding environment and contextual clues. Nearby car is more likely to provide enough contextual information for the model to recognize due to its larger size.

Object size and resolution: Nearby objects occupy more pixels in the image, so the key features of these objects (e.g., edges, shape, color, etc.) remain relatively legible even with added noise. In contrast, distant trucks have lower resolution in the image due to their smaller size, and noise is more likely to mask their key features, leading to recognition difficulties.

This is all I have learned from all of this PROJECT and I will continue to explore it in the future.
